### DISCUSSION TITLE
in the same java application . opentelemty-sdk create span manully and opentelemetry -java-instrument create span by instrument can work at the same time ?

### DISCUSSION THREAD

**Discussion by chenyi19851209** [open-telemetry/opentelemetry-java#3516] [Q&A]
please ask: 

    in the same java application . opentelemty-sdk create span manully and opentelemetry -java-instrument create span by instrument can work at the same time  ?

demo:  springboot restful controller start  request  DB （use opentelemetry -java-instrument）， request redis（use opentelemetry -java-instrument）and **request  customize rpc framework （use opentelemetry -java）**

please ask can they use same traceId?

thanks

**Comment by jkwatson** [comment]
You can definitely create spans manually in addition to the ones created by the auto-instrumentation agent. 
Documentation for this is here:  https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/manual-instrumentation.md

**Reply by chenyi19851209** [reply]
thanks for your answer.
i read you given document, i think this method can work when auto-instrumentation installed.but i expect another condition:
our provider sdk track for our customize rpc framework before. and now our hope transfer to opentelement ,and aims to reduce javaAgent dangerous. our hope   our customize rpc framework can work without auto-instrumentation(use sdk). and the at same time  application can use auto-instrument when application use common http or db request.

thanks for your answer.

**Reply by anuraaga** [reply]
@chenyi19851209 The terminology is a bit complicated, but you can think of there being three components, the OpenTelemetry API (opentelemetry-api), SDK (opentelemetry-sdk), and auto-instrumentation (the confusion can come up since we often refer to both the API+SDK together as "the SDK"). When you write the instrumentation for your custom RPC framework, you must only use opentelemetry-api, instrumentation never depends on opentelemetry-sdk (this is true for the instrumentation used by auto instrumentation as well). As long as you use opentelemetry-api, then it will work fine either with auto instrumentation, or if an end-user app (not the RPC framework) includes and initializes opentelemetry-sdk manually.

**Reply by chenyi19851209** [reply]
thanks for your answer.
 so i understand it will work fine in my condition. i will try for test.
